### PR TITLE
Added login with password encrypted

### PR DESCRIPTION
**Pull request by darkvergus**
- Added EncryptionHelper to help with encryption.
- Added basic password validation for the login.
- Right now its not possible to just connect with username as before.

## Summary by CodeRabbit

- **New Features**
	- Introduced a required `Password` property for account management.
	- Added methods for password verification and updating machine IDs to enhance account security.
	- Enhanced login functionality with a new command to handle login requests, including username and password verification.
	- Improved security by logging only the username during login attempts.

- **Bug Fixes**
	- Updated login logic to provide clearer error responses for empty usernames and passwords.

- **Documentation**
	- Improved encapsulation and security measures for user authentication processes.

### CODE REVIEW

File: Maple2.Server.World/Service/GlobalService.cs [hunk 1]
```diff
@@ -55,14 +69,20 @@ public override Task<LoginResponse> Login(LoginRequest request, ServerCallContex
...
+        if (!isPasswordValid) {
+            return Task.FromResult(new LoginResponse {
+                Code = LoginResponse.Types.Code.ErrorPassword,
+                Message = "Incorrect Password."
+            });
+        }
 
         if (account.MachineId != machineId) {
-            logger.Warning("MachineId mismatch for account {AccountId}", account.Id);
+            logger.Warning($"MachineId mismatch for account {account.Id}");
```
Comment (AngeloTadeucci) [inline Maple2.Server.World/Service/GlobalService.cs:81]: for logging its not recommended using interpolation

**Review by AngeloTadeucci — CHANGES_REQUESTED** [review]

File: Maple2.Model/Game/User/Account.cs [hunk 2]
```diff
@@ -29,6 +30,7 @@ public class Account {
     public int SurvivalSilverLevelRewardClaimed { get; set; }
     public int SurvivalGoldLevelRewardClaimed { get; set; }
     public bool ActiveGoldPass { get; set; }
+    public byte[] Salt { get; set; }
```
Comment (AngeloTadeucci) [inline Maple2.Model/Game/User/Account.cs:33]: not being used?

Comment (darkvergus) [reply Maple2.Model/Game/User/Account.cs:33]: did i not remove it? wth

Comment (darkvergus) [reply Maple2.Model/Game/User/Account.cs]: done

File: Maple2.Server.World/Service/GlobalService.cs (hunk 1 above)
Comment (darkvergus) [reply Maple2.Server.World/Service/GlobalService.cs]: done

**Review by AngeloTadeucci — APPROVED** [review]

File: Maple2.Model/Game/User/Account.cs [hunk 3]
```diff
@@ -9,6 +9,7 @@ public class Account {
     public long Id { get; init; }
 
     public required string Username { get; init; }
+    public required string Password { get; init; }
```
Comment (Zintixx) [inline Maple2.Model/Game/User/Account.cs:12]: Do we need Password on this Account? I would figure it's only used on DB Model. Once a user has logged in, it's no longer necessary.

Comment (darkvergus) [reply Maple2.Model/Game/User/Account.cs:12]: its in use inside GlobalService

**Review by AngeloTadeucci — APPROVED** [review]

**Review by Zintixx — APPROVED** [review]

File: Maple2.Server.World/Service/GlobalService.cs [hunk 4]
```diff
@@ -55,21 +65,26 @@ public override Task<LoginResponse> Login(LoginRequest request, ServerCallContex
             });
         }
 
-        // TODO: Implement password check
+        bool isPasswordValid = db.VerifyPassword(account.Id, password);
+        if (!isPasswordValid) {
+            return Task.FromResult(new LoginResponse {
+                Code = LoginResponse.Types.Code.ErrorPassword,
+                Message = "Incorrect Password."
```
Comment (Zintixx) [inline Maple2.Server.World/Service/GlobalService.cs:72]: comma here too

File: Maple2.Server.World/Service/GlobalService.cs [hunk 5]
```diff
@@ -22,12 +21,23 @@ public GlobalService(GameStorage gameStorage) {
...
+            return Task.FromResult(new LoginResponse {
+                Code = LoginResponse.Types.Code.ErrorId,
+                Message = "Invalid Id"
+            });
+        }
+        
+        if (string.IsNullOrWhiteSpace(request.Password)) {
+            return Task.FromResult(new LoginResponse {
+                Code = LoginResponse.Types.Code.ErrorPassword,
+                Message = "Invalid Password."
```
Comment (Zintixx) [inline Maple2.Server.World/Service/GlobalService.cs:33]: comma

**Review by Zintixx — CHANGES_REQUESTED** [review]
just some tiny formatting things

**Review by Zintixx — APPROVED** [review]

**Pull request merged**